src/frontend/js/components/Modal/index.tsx

```diff
@@ -43,6 +66,23 @@ export const Modal = ({
       overlayClassName={mergeClasses({ base: 'modal__overlay', classes: overlayClassName })}
       {...props}
     >
+      <div className={headerClasses.join(' ')}>
```

Comment:
In order to improve semantic what do you think to use a `header` tag here ?
```suggestion
      <header className={headerClasses.join(' ')}>
```